### PR TITLE
Rating Identifier already exists tests

### LINKED ISSUE #298: [C4GT] Network-Score: Add tests for 'RatingIdentifierAlreadyAdded'
## Description
Subtask under : https://github.com/cord-network/community/issues/7

New testcase can be tested under `cargo test -p pallet-network-score` after adding it in the code.

## Goals

- [ ] Add tests for `RatingIdentifierAlreadyAdded` for pallet/network-score

## Expected Outcome
- [ ] Test should assert for `RatingIdentifierAlreadyAdded` being returned properly in all the possible calls.
- [ ] If there are more than one function returning this error code, all of them should be validated either as one test case or as multiple test cases.
       
## Acceptance Criteria
NA

## Implementation Details
Look at other test cases, and add a test case for the same.
Ex: In pallets/network-membership check method `test_duplicate_member_request` which validates error code `MembershipAlreadyAcquired`

## Mockups / Wireframes
NA

---- 

## Product Name
CORD

## Organization Name
Dhiway

## Domain
Blockchain

## Tech Skills Needed
Rust

## Mentor(s)
@amarts 

## Complexity
[Low]

## Category
[Test]

## Sub Category
[Beginner friendly]

### PR DESCRIPTION
**Pull request by zeel991**
- Fixes https://github.com/dhiway/cord/issues/298
In this test , rating is registered once  , and when rating is registered again , test uses different `message id` but same `digest` , as using the same message id gives `MessageIdentifierAlreadyExists`
Thus , if the rating Id is already in use , it will give the error `RatingIdentifierAlreadyAdded`
Please give your feedback if I have made any mistake at some point @vatsa287 
Thank you .

### CODE REVIEW

**Comment by zeel991** [comment]
Hey @vatsa287 , During my previous attempt of writing this test i wrote a test but it gave `messageidalreadyexists` test , so to avoid that I used different message id her , but it is showing some error , can you please just let me know why is the test panicking  , I'll try and solve the issue from your advice .

**Comment by vatsa287** [comment]
@zeel991 Will review it today

File: pallets/network-score/src/tests.rs [hunk 1]
```diff
@@ -159,3 +159,67 @@ fn check_duplicate_message_id() {
...
+		assert_ok!(Score::register_rating(
+			DoubleOrigin(author.clone(), creator.clone()).into(),
+			entry.clone(),
+			entry_digest,
+			message_id.clone(),
+			authorization_id.clone(),
+		));
+
+		// Attempt to register another rating entry with the same identifier
+		assert_err!(
```
Comment (vatsa287) [inline pallets/network-score/src/tests.rs:303]: Please change to
```
		// Remove message_id and provider_did from entries
		<MessageIdentifiers<Test>>::remove(message_id.clone(), creator.clone());

		assert_err!(Score::register_rating(
			DoubleOrigin(author.clone(), creator.clone()).into(),
			entry.clone(),
			entry_digest,
			message_id.clone(),
			authorization_id.clone(),
			),
			Error::<Test>::RatingIdentifierAlreadyAdded
		);
```

**Review by vatsa287 — CHANGES_REQUESTED** [review]
This change is required because there is no other way to reach that block of code, without making explicit change since changing the entry with new `message_id`, or `creator` would create a new `identifier` whereas we require the same `identifier` to be present.

**Comment by zeel991** [comment]
Thank you, @vatsa287. I had this approach in the back of my mind, but somehow got confused while implementing it. The test has passed all three tests. If everything seems alright, please assign me the issue before merging the PR.

**Review by vatsa287 — DISMISSED** [review]
LGTM

**Comment by zeel991** [comment]
@amarts , Please review the PR and give your feedback on it .
Thank you

**Review by amarts — DISMISSED** [review]
LGTM

**Review by amarts — APPROVED** [review]

**Pull request merged**